app/views/welcome/support_us.html.slim

```diff
@@ -1,58 +1,90 @@
-        input name="on0" type="hidden" value="One time show of support" /
-        input name="on1" type="hidden" value="Your Twitter / GitHub username" /
-        input name="currency_code" type="hidden" value="SGD" /
-        .row
-          .input-field.col.s12
-            select#one_time_support name="os0"
-              option value="Pat on the back"  Pat on the back $50.00 SGD
-              option value="Awesome Job"  Awesome Job $150.00 SGD
-              option value="True Believer"  True Believer $300.00 SGD
-              option value="Shining glory of the community"  Shining glory of the community $900.00 SGD
-            label for="one_time_support" One time show of support
-        .row
-          .input-field.col.s12
-            input#user_details2 maxlength="200" name="os1" type="text" /
-            label for ="user_details2" Your Twitter / GitHub username
-        .row
-          .input-field.col.s12.center-align
-            input alt="PayPal – The safer, easier way to pay online!" border="0" name="submit" src="https://www.paypalobjects.com/en_GB/SG/i/btn/btn_paynowCC_LG.gif" type="image" /
-            img alt="" border="0" height="1" src="https://www.paypalobjects.com/en_GB/i/scr/pixel.gif" width="1" /
+    - support_plans.keys.each do |plan|
+      .col.s12.l5
+
+        h5 = support_plans[plan][:title]
+
+        form
+          .row
+            .input-field.col.s12
+              select id="#{support_plans[plan][:id]}" name="os0"
+                - support_plans[plan][:payment_options].each do |payment_option|
+                  option value="#{payment_option[:stripe_code]}" = payment_option[:label]
+              label id="#{support_plans[plan][:id]}" = support_plans[plan][:label]
+          .row
+            .input-field.col.s12
+              label for="user_details" Your Email
+              input.user_details maxlength="200" name="os1" type="email"
+          .row.center-align
+            .input-field.btn.waves-effect.waves-light.pay-with-stripe type="button" Pay with Stripe
+
+
+javascript:
+  (function() {
+    const stripe = Stripe('pk_test_cnydRVmI6aUKzk7GOr4v8QqT');
+
+    const checkoutButtons = document.getElementsByClassName('pay-with-stripe');
+
+    for (let i = 0; i < checkoutButtons.length; i++) {
+      checkoutButtons[i].addEventListener('click', () => {
+        const chosenOption = document.getElementsByTagName('select')[i].value;
+        const customerEmail = document.getElementsByClassName('user_details')[i].value;
+
+        const productType = chosenOption.includes('plan_') ? 'plan' : 'sku';
+
+        const checkoutOptions = {
+          items: [{ [productType]: chosenOption, quantity: 1}],
+          successUrl: 'https://success.com',
+          cancelUrl: 'https://canceled.com'
+        }
+
+        customerEmail ? checkoutOptions["customerEmail"] = customerEmail : null;
 
+        stripe.redirectToCheckout(checkoutOptions)
+        .then(function (result) {
+          if (result.error) {
```

Comment:
What do we wanna do here?

Reply:
discussing options here: https://github.com/engineersftw/website/issues/55#issuecomment-571467676

Reply:
This is very rare (e.g. when connection drops etc.) Maybe we can just do an alert with the error message?